src/main.rs

```diff
@@ -20,6 +20,15 @@ fn positive_integer(val: String) -> Result<(), String> {
     u32::from_str(&val).map(|_| ()).map_err(|e| format!("{e}"))
 }
 
+fn label_list(val: String) -> Result<(), String> {
+    val.chars()
+        .all(|c| {
+            c == ',' || (c.is_ascii() && c.is_alphanumeric()) || c == '-' || c == '_' || c == '.'
```

Comment:
This could be implemented using the `u8` functions instead (and `as_bytes().iter()`) to avoid the need for `is_ascii`. Also, `c.is_alphanumeric() || c == '-' || c == '_' || c == '.'` seems like it should be a function defined in the parser somewhere...

Reply:
I simply pulled the parser's `is_valid_label` which indeed does exactly what we want here.